tests/testGradingMethods.js

```diff
@@ -0,0 +1,459 @@
+            questionsPage = await (await fetch(iqUrl)).text();
+            $questionsPage = cheerio.load(questionsPage);
+          }
+        );
+        after('close external grader socket', () => socket.close());
+
+        it('should result in 1 grading jobs', async () => {
+          const grading_jobs = (await sqlDb.queryAsync(sql.get_grading_jobs_by_iq, { iqId })).rows;
+          assert.lengthOf(grading_jobs, 1);
+        });
+        it('should result in 1 "pastsubmission-block" component being rendered', () => {
+          assert.lengthOf($questionsPage('.pastsubmission-block'), 1);
+        });
+        it('should be given submission grade in "pastsubmission-block"', async () => {
+          assert.include(
+            questionsPage,
+            '<td>Awarded points:</td>\n          <td>\n\n\n<span class="badge badge-danger">\n\n0/6\n</span>'
+          );
+        });
+        it('should result in 1 "grading-block" component being rendered', () => {
+          assert.lengthOf($questionsPage('.grading-block'), 0);
+        });
+      });
+      describe('"save" action', () => {
+        before(
+          'load page as student and submit "grade" action to "External" type question',
+          async () => {
+            const hm1Body = await loadHomeworkPage(mockStudents[1]);
+            $hm1Body = cheerio.load(hm1Body);
+            iqUrl =
+              siteUrl +
+              $hm1Body(
+                'a:contains("HW9.3. External Grading: Fibonacci function, file upload")'
+              ).attr('href');
+
+            gradeRes = await saveOrGrade(iqUrl, {}, 'save', [
+              { name: 'fib.py', contents: Buffer.from(anyFileContent).toString('base64') },
+            ]);
+            assert.equal(gradeRes.status, 200);
+
+            questionsPage = await gradeRes.text();
+            $questionsPage = cheerio.load(questionsPage);
+
+            iqId = parseInstanceQuestionId(iqUrl);
+          }
+        );
+        it('should NOT result in any grading jobs', async () => {
+          const grading_jobs = (await sqlDb.queryAsync(sql.get_grading_jobs_by_iq, { iqId })).rows;
+          assert.lengthOf(grading_jobs, 0);
+        });
+        it('should result in 1 "pastsubmission-block" component being rendered', () => {
+          assert.lengthOf($questionsPage('.pastsubmission-block'), 1);
+        });
+        it('should NOT be given submission grade in "pastsubmission-block"', async () => {
+          assert.notInclude(
+            questionsPage,
+            'Submitted answer\n          \n        </span>\n        <span>\n    \n        <span class="badge badge-success">correct: 100%</span>'
+          );
+        });
+        it('should NOT result in "grading-block" component being rendered', () => {
+          assert.lengthOf($questionsPage('.grading-block'), 0);
+        });
+      });
+    });
+  });
+
+  //manual-grading-col-split branch WILL INTRODUCE LOGIC FOR GRADING TYPES ARRAY CONFIGURATION
```

Comment:
I'm going to opt to remove this from this branch so that master stays clean. When https://github.com/PrairieLearn/PrairieLearn/pull/3835 is merged, that should bring these tests with it.

Reply:
Done in d990a75.